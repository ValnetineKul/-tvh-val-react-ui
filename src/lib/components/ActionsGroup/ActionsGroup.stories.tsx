import { action } from '@storybook/addon-actions';
import React from 'react';
import { createMeta, createTemplate } from '../../story-utils';
import Button from '../Buttons/Button';
import IconButton from '../Buttons/IconButton';
import ListIconButton from '../Buttons/ListIconButton';
import Icon from '../Icon';
import { Print } from '../Icon/icons/functional';
import MenuItem from '../Menus/MenuItem';
import ActionsGroup from './ActionsGroup';

const Template = createTemplate(ActionsGroup);

export const PrimitiveTwo = Template.bind({});
PrimitiveTwo.args = {
  buttons: [
    ({ isFullWidth }) => <Button variant="primary" label="Add to basket" fullWidth={isFullWidth} />,
    ({ isFullWidth }) => <Button variant="tertiary" label="Print labels" fullWidth={isFullWidth} />,
  ],
  visibleAmount: 3,
};
PrimitiveTwo.storyName = 'ActionsGroup-two';

export const PrimitiveTwoWithIcon = Template.bind({});
PrimitiveTwoWithIcon.args = {
  buttons: [({ isFullWidth }) => <Button variant="primary" label="Add to basket" fullWidth={isFullWidth} />],
  iconButtons: [() => <ListIconButton onClick={action('Add to list')} />],
};
PrimitiveTwoWithIcon.storyName = 'ActionsGroup-two-icon';

export const PrimitiveThree = Template.bind({});
PrimitiveThree.args = {
  buttons: [
    ({ isFullWidth }) => <Button variant="primary" label="Add to basket" fullWidth={isFullWidth} />,
    ({ isFullWidth, isSquashed }) =>
      isSquashed ? (
        <MenuItem label="Print labels" />
      ) : (
        <Button variant="tertiary" label="Print labels" fullWidth={isFullWidth} />
      ),
  ],
  iconButtons: [
    ({ isSquashed }) =>
      isSquashed ? <MenuItem label="Add to list" /> : <ListIconButton onClick={action('Add to list')} />,
  ],
  visibleAmount: 3,
};
PrimitiveThree.storyName = 'ActionsGroup-three';

export const PrimitiveFour = Template.bind({});
PrimitiveFour.args = {
  buttons: [
    ({ isFullWidth }) => <Button variant="primary" label="Add to basket" fullWidth={isFullWidth} />,
    ({ isFullWidth, isSquashed }) =>
      isSquashed ? (
        <MenuItem label="Print labels" />
      ) : (
        <Button variant="tertiary" label="Print labels" fullWidth={isFullWidth} />
      ),
    ({ isFullWidth, isSquashed }) =>
      isSquashed ? (
        <MenuItem label="Recalculate" />
      ) : (
        <Button variant="tertiary" label="Recalculate" fullWidth={isFullWidth} />
      ),
  ],
  iconButtons: [
    ({ isSquashed }) =>
      isSquashed ? <MenuItem label="Add to list" /> : <ListIconButton onClick={action('Add to list')} />,
  ],
  visibleAmount: 3,
};
PrimitiveFour.storyName = 'ActionsGroup-four';

export const PrimitiveFive = Template.bind({});
PrimitiveFive.args = {
  buttons: [
    ({ isFullWidth }) => <Button variant="primary" label="Add to basket" fullWidth={isFullWidth} />,
    ({ isFullWidth, isSquashed }) =>
      isSquashed ? (
        <MenuItem label="Print labels" />
      ) : (
        <Button variant="tertiary" label="Print labels" fullWidth={isFullWidth} />
      ),
    ({ isFullWidth, isSquashed }) =>
      isSquashed ? (
        <MenuItem label="Recalculate" />
      ) : (
        <Button variant="tertiary" label="Recalculate" fullWidth={isFullWidth} />
      ),
    ({ isFullWidth, isSquashed }) =>
      isSquashed ? <MenuItem label="Remove" /> : <Button variant="tertiary" label="Remove" fullWidth={isFullWidth} />,
  ],
  iconButtons: [
    ({ isSquashed }) =>
      isSquashed ? <MenuItem label="Add to list" /> : <ListIconButton onClick={action('Add to list')} />,
  ],
  visibleAmount: 3,
};
PrimitiveFive.storyName = 'ActionsGroup-five';

export const PrimitiveMore = Template.bind({});
PrimitiveMore.args = {
  buttons: [
    ({ isFullWidth }) => <Button variant="primary" label="Add to basket" fullWidth={isFullWidth} />,
    ({ isFullWidth, isSquashed }) =>
      isSquashed ? (
        <MenuItem label="Print labels" />
      ) : (
        <Button variant="tertiary" label="Print labels" fullWidth={isFullWidth} />
      ),
    ({ isFullWidth, isSquashed }) =>
      isSquashed ? (
        <MenuItem label="Recalculate" />
      ) : (
        <Button variant="tertiary" label="Recalculate" fullWidth={isFullWidth} />
      ),
    ({ isFullWidth, isSquashed }) =>
      isSquashed ? <MenuItem label="Remove" /> : <Button variant="tertiary" label="Remove" fullWidth={isFullWidth} />,
  ],
  iconButtons: [
    ({ isSquashed }) =>
      isSquashed ? <MenuItem label="Add to list" /> : <ListIconButton onClick={action('Add to list')} />,
    ({ isSquashed }) =>
      isSquashed ? <MenuItem label="Print" /> : <IconButton onClick={action('Print')} icon={<Icon icon={Print} />} />,
  ],
  visibleAmount: 3,
};
PrimitiveMore.storyName = 'ActionsGroup-six';

export default createMeta({
  title: 'Components/ActionsGroups/ActionsGroup',
  component: ActionsGroup,
});
