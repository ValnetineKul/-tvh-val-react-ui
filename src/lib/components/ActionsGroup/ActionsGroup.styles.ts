import { makeStyles } from '../../../themes/core';

const useStyles = makeStyles({ name: 'ActionsGroup' })((theme) => ({
  root: {
    display: 'flex',
    flex: 1,
    margin: 0,
    padding: 0,
    flexWrap: 'wrap',
    alignItems: 'center',
    justifyContent: 'flex-start',
    gap: theme.layoutSpacing['spacing/400'],
  },

  reversedList: {
    flexDirection: 'row-reverse',
    justifyContent: 'flex-start',
  },

  startAlinment: {
    justifyContent: 'flex-start',
  },

  actionWrapper: {
    listStyle: 'none',
  },

  buttonWrapperMobile: {
    width: '100%',
  },

  lastButtonWrapperMobile: {
    flex: '1 1',
  },
}));

export default useStyles;
