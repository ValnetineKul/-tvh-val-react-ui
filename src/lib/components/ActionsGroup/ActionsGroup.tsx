import type { FC, ReactElement } from 'react';
import React, { useContext, useEffect, useMemo } from 'react';

import OverflowAction from '../../patterns/OverflowAction';
import useStyles from './ActionsGroup.styles';
import useScreenSize from '../../hooks/useScreenSize';
import { HasSingleIconButtonContext } from '../../patterns/Headers/SectionHeader/HasSingleIconButtonContext/HasSingleIconButtonContext';

export type RenderAction = (props: { isFullWidth?: boolean; isSquashed?: boolean }) => ReactElement;

export interface ActionsGroupProps {
  buttons?: (RenderAction | undefined)[];
  iconButtons?: (RenderAction | undefined)[];
  visibleAmount?: 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9;
  className?: string;
  mobileTitle?: string;
}

const ActionsGroup: FC<ActionsGroupProps> = ({
  buttons = [],
  iconButtons = [],
  visibleAmount = 3,
  className,
  mobileTitle,
}) => {
  const { setHasSingleIconButton } = useContext(HasSingleIconButtonContext);
  const { classes, cx } = useStyles();
  const { isTablet, isTabletUp } = useScreenSize('Tablet');
  const isMobile = !isTabletUp;
  const actualVisibleAmount = !isTabletUp || isTablet ? Number(visibleAmount) : Number(visibleAmount) + 1;

  useEffect(() => {
    if (setHasSingleIconButton) {
      setHasSingleIconButton(buttons.length === 0 && iconButtons.length === 1);
    }
  }, [buttons, iconButtons, setHasSingleIconButton]);

  const actions = useMemo(() => {
    const actualButtons = buttons.filter((b) => b);
    const actualIconButtons = iconButtons.filter((ib) => ib);
    const hasOverflow = actualButtons.length + actualIconButtons.length > actualVisibleAmount;

    const { menuItems, actionButtons } = actualButtons
      .concat([...actualIconButtons].reverse())
      .reduce<{ actionButtons: ReactElement[]; menuItems: ReactElement[] }>(
        (acc, renderButton: RenderAction, index) => {
          if (hasOverflow && index + 1 > actualVisibleAmount - 1) {
            acc.menuItems = acc.menuItems.concat(renderButton({ isSquashed: true }));
            return acc;
          }
          acc.actionButtons = acc.actionButtons.concat(
            <li
              className={cx(classes.actionWrapper, {
                [classes.buttonWrapperMobile]: isMobile && actualButtons.length > index,
                [classes.lastButtonWrapperMobile]: hasOverflow
                  ? isMobile && actualVisibleAmount - 1 === index + 1
                  : isMobile && actualButtons.length - 1 === index,
              })}
              key={`action-${index}`}
            >
              {renderButton({ isFullWidth: isMobile })}
            </li>
          );
          return acc;
        },
        {
          menuItems: [],
          actionButtons: [],
        }
      );

    return (
      <>
        {actionButtons}
        {menuItems.length > 0 && (
          <li className={classes.actionWrapper}>
            <OverflowAction menuItems={menuItems} mobileMenuTitle={mobileTitle} />
          </li>
        )}
      </>
    );
    // eslint-disable-next-line
  }, [isMobile, buttons, iconButtons, cx, classes, actualVisibleAmount]);

  return (
    <ul
      className={cx(
        classes.root,
        {
          [classes.startAlinment]: isMobile,
          [classes.reversedList]: !isTablet && isTabletUp,
        },
        className
      )}
    >
      {actions}
    </ul>
  );
};
export default ActionsGroup;
