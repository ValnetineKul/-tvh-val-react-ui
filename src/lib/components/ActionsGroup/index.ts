import ActionsGroup from './ActionsGroup';

export type { RenderAction, ActionsGroupProps } from './ActionsGroup';

export default ActionsGroup;
