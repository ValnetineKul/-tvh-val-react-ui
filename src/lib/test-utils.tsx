import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
