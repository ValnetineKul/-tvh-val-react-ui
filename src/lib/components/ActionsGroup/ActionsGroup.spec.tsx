import React from 'react';
import { initResponsiveTest, render, screen, userEvent } from '../../test-utils';
import type { RenderAction } from './ActionsGroup';
import ActionsGroup from './ActionsGroup';

interface Params {
  buttonsCount?: number;
  iconButtonsCount?: number;
  visibleAmount?: 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9;
  className?: string;
}

const getProps = (params?: Params) => {
  const { buttonsCount = 2, iconButtonsCount = 1, visibleAmount, className } = params || {};

  const buttons: RenderAction[] = [];
  const iconButtons: RenderAction[] = [];

  for (let i = 0; i < buttonsCount; i++) {
    buttons.push(({ isFullWidth, isSquashed }) =>
      isSquashed ? (
        <li key={`item-${i}`}>{`MenuItem ${i + 1}`}</li>
      ) : (
        <button className={isFullWidth ? 'full' : ''} type="button">
          {`Button ${i + 1}`}
        </button>
      )
    );
  }
  for (let i = 0; i < iconButtonsCount; i++) {
    iconButtons.push(({ isSquashed }) =>
      isSquashed ? (
        <li key={`icon-item-${i}`}>{`IconMenuItem ${i + 1}`}</li>
      ) : (
        <button type="button">{`IconButton ${i + 1}`}</button>
      )
    );
  }

  return {
    buttons,
    className,
    iconButtons,
    visibleAmount,
  };
};

describe('ActionsGroup', () => {
  it('Should render correctly on the Mobile', () => {
    initResponsiveTest('Mobile');
    const props = getProps();
    const { container } = render(<ActionsGroup {...props} />);
    expect(container.firstChild).toMatchSnapshot('ActionsGroup mobile');
  });
  it('Should render correctly on the Tablet', () => {
    initResponsiveTest('Tablet');
    const props = getProps();
    const { container } = render(<ActionsGroup {...props} />);
    expect(container.firstChild).toMatchSnapshot('ActionsGroup tablet');
  });
  it('Should render correctly on the Desktop', () => {
    initResponsiveTest('Desktop');
    const props = getProps();
    const { container } = render(<ActionsGroup {...props} />);
    expect(container.firstChild).toMatchSnapshot('ActionsGroup desktop');
  });

  it('Should use correct className', () => {
    const props = getProps({ className: 'custom-class' });
    const { container } = render(<ActionsGroup {...props} />);

    expect(container.firstChild).toHaveClass(props.className as string);
  });

  it('Buttons should be fullWidth on mobile and the last one should be flexible', () => {
    const props = getProps();
    initResponsiveTest('Mobile');
    render(<ActionsGroup {...props} />);
    const button1 = screen.getByText('Button 1', { selector: 'button' });
    const button2 = screen.getByText('Button 2', { selector: 'button' });
    const iconButton = screen.getByText('IconButton 1', { selector: 'button' });

    expect(button1).toHaveClass('full');
    expect(button2).toHaveClass('full');
    expect(button1.parentElement?.className).not.toMatch(/lastButtonWrapperMobile/);
    expect(button2.parentElement?.className).toMatch(/lastButtonWrapperMobile/);
    expect(iconButton).not.toHaveClass('full');
  });

  it('Should render actions in the direct order on the mobile', () => {
    const props = getProps();
    initResponsiveTest('Mobile');
    const { firstChild } = render(<ActionsGroup {...props} />).container;
    const button1 = screen.getByText('Button 1', { selector: 'button' });
    const button2 = screen.getByText('Button 2', { selector: 'button' });
    const iconButton = screen.getByText('IconButton 1', { selector: 'button' });

    expect(firstChild?.childNodes[0]).toContainElement(button1);
    expect(firstChild?.childNodes[1]).toContainElement(button2);
    expect(firstChild?.childNodes[2]).toContainElement(iconButton);
  });

  it('Should render actions in the direct order on the tablet', () => {
    const props = getProps();
    initResponsiveTest('Tablet');
    const { firstChild } = render(<ActionsGroup {...props} />).container;
    const button1 = screen.getByText('Button 1', { selector: 'button' });
    const button2 = screen.getByText('Button 2', { selector: 'button' });
    const iconButton = screen.getByText('IconButton 1', { selector: 'button' });

    expect(firstChild?.childNodes[0]).toContainElement(button1);
    expect(firstChild?.childNodes[1]).toContainElement(button2);
    expect(firstChild?.childNodes[2]).toContainElement(iconButton);
  });

  it('Should render actions in the reversed order on the desktop', () => {
    const props = getProps({ buttonsCount: 3 });
    initResponsiveTest('Desktop');
    const { firstChild, firstElementChild } = render(<ActionsGroup {...props} />).container;
    const button1 = screen.getByText('Button 1', { selector: 'button' });
    const button2 = screen.getByText('Button 2', { selector: 'button' });
    const button3 = screen.getByText('Button 3', { selector: 'button' });
    const iconButton = screen.getByText('IconButton 1', { selector: 'button' });

    expect(firstChild?.childNodes[0]).toContainElement(button1);
    expect(firstChild?.childNodes[1]).toContainElement(button2);
    expect(firstChild?.childNodes[2]).toContainElement(button3);
    expect(firstChild?.childNodes[3]).toContainElement(iconButton);
    expect(firstElementChild?.className).toMatch(/reversedList/);
  });

  it('Should render overflow with the correct ordering if the actions amount is more than visibleAmount on mobile', () => {
    const props = getProps({ buttonsCount: 2, visibleAmount: 2 });
    initResponsiveTest('Mobile');
    render(<ActionsGroup {...props} />);

    expect(screen.getByText('Button 1', { selector: 'button' })).toBeInTheDocument();
    expect(screen.queryByText('Button 2', { selector: 'button' })).not.toBeInTheDocument();
    expect(screen.queryByText('IconButton 1', { selector: 'button' })).not.toBeInTheDocument();

    const overflowButton = screen.getByText('ellipsisH.svg', { selector: 'svg' });
    expect(overflowButton).toBeInTheDocument();
    userEvent.click(overflowButton);

    const menuItem2 = screen.getByText('MenuItem 2', { selector: 'li' });
    const dropDown = menuItem2.parentElement;
    const iconButton = screen.getByText('IconMenuItem 1', { selector: 'li' });

    expect(dropDown?.childNodes[0]).toContainElement(menuItem2);
    expect(dropDown?.childNodes[1]).toContainElement(iconButton);
  });

  it('Should render overflow with the correct ordering if the actions amount is more than visibleAmount on tablet', () => {
    const props = getProps({ buttonsCount: 2, visibleAmount: 2 });
    initResponsiveTest('Tablet');
    render(<ActionsGroup {...props} />);

    expect(screen.getByText('Button 1', { selector: 'button' })).toBeInTheDocument();
    expect(screen.queryByText('Button 2', { selector: 'button' })).not.toBeInTheDocument();
    expect(screen.queryByText('IconButton 1', { selector: 'button' })).not.toBeInTheDocument();

    const overflowButton = screen.getByText('ellipsisH.svg', { selector: 'svg' });
    expect(overflowButton).toBeInTheDocument();
    userEvent.click(overflowButton);

    const menuItem2 = screen.getByText('MenuItem 2', { selector: 'li' });
    const dropDown = menuItem2.parentElement;
    const iconButton = screen.getByText('IconMenuItem 1', { selector: 'li' });

    expect(dropDown?.childNodes[0]).toContainElement(menuItem2);
    expect(dropDown?.childNodes[1]).toContainElement(iconButton);
  });

  it('Should render overflow with the correct ordering if the actions amount is more than visibleAmount on desktop', () => {
    const props = getProps({ buttonsCount: 3, visibleAmount: 2 });
    initResponsiveTest('Desktop');
    render(<ActionsGroup {...props} />);

    expect(screen.getByText('Button 1', { selector: 'button' })).toBeInTheDocument();
    expect(screen.getByText('Button 2', { selector: 'button' })).toBeInTheDocument();
    expect(screen.queryByText('Button 3', { selector: 'button' })).not.toBeInTheDocument();
    expect(screen.queryByText('IconButton 1', { selector: 'button' })).not.toBeInTheDocument();

    const overflowButton = screen.getByText('ellipsisH.svg', { selector: 'svg' });
    expect(overflowButton).toBeInTheDocument();
    userEvent.click(overflowButton);

    const menuItem3 = screen.getByText('MenuItem 3', { selector: 'li' });
    const dropDown = menuItem3.parentElement;
    const iconButton = screen.getByText('IconMenuItem 1', { selector: 'li' });

    expect(dropDown?.childNodes[0]).toContainElement(menuItem3);
    expect(dropDown?.childNodes[1]).toContainElement(iconButton);
  });
});
