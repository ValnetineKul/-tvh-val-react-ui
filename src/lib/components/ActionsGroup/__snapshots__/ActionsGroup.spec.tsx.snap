// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionsGroup Should render correctly on the Desktop: ActionsGroup desktop 1`] = `
<ul
  class="tss-y1q3yg-ActionsGroup-root-ActionsGroup-reversedList"
>
  <li
    class="tss-aagoa-ActionsGroup-actionWrapper"
  >
    <button
      class=""
      type="button"
    >
      Button 1
    </button>
  </li>
  <li
    class="tss-aagoa-ActionsGroup-actionWrapper"
  >
    <button
      class=""
      type="button"
    >
      Button 2
    </button>
  </li>
  <li
    class="tss-aagoa-ActionsGroup-actionWrapper"
  >
    <button
      type="button"
    >
      IconButton 1
    </button>
  </li>
</ul>
`;

exports[`ActionsGroup Should render correctly on the Mobile: ActionsGroup mobile 1`] = `
<ul
  class="tss-1n9rhgh-ActionsGroup-root-ActionsGroup-startAlinment"
>
  <li
    class="tss-11kknj0-ActionsGroup-actionWrapper-ActionsGroup-buttonWrapperMobile"
  >
    <button
      class="full"
      type="button"
    >
      Button 1
    </button>
  </li>
  <li
    class="tss-1h74icw-ActionsGroup-actionWrapper-ActionsGroup-buttonWrapperMobile-ActionsGroup-lastButtonWrapperMobile"
  >
    <button
      class="full"
      type="button"
    >
      Button 2
    </button>
  </li>
  <li
    class="tss-aagoa-ActionsGroup-actionWrapper"
  >
    <button
      type="button"
    >
      IconButton 1
    </button>
  </li>
</ul>
`;

exports[`ActionsGroup Should render correctly on the Tablet: ActionsGroup tablet 1`] = `
<ul
  class="tss-5cypwz-ActionsGroup-root"
>
  <li
    class="tss-aagoa-ActionsGroup-actionWrapper"
  >
    <button
      class=""
      type="button"
    >
      Button 1
    </button>
  </li>
  <li
    class="tss-aagoa-ActionsGroup-actionWrapper"
  >
    <button
      class=""
      type="button"
    >
      Button 2
    </button>
  </li>
  <li
    class="tss-aagoa-ActionsGroup-actionWrapper"
  >
    <button
      type="button"
    >
      IconButton 1
    </button>
  </li>
</ul>
`;
